Document APIs for the System.Text.Encodings.Web Namespace.

tompratt-AQ opened the pull request:
# Title
Document APIs for the System.Text.Encodings.Web Namespace.

## Summary
Added documentation for "To be added." fields in all classes of the System.Text.Encodings.Web Namespace:

- System.Text.Encodings.Web.HtmlEncoder
- System.Text.Encodings.Web.JavaScriptEncoder
- System.Text.Encodings.Web.TextEncoder
- System.Text.Encodings.Web.TextEncoderSettings
- System.Text.Encodings.Web.UrlEncoder

Fixes #2211 

## Suggested Reviewers
@mairaw

tompratt-AQ committed:
Document APIs for the System.Text.Encodings.Web Namespace.

tompratt-AQ committed:
Fix build error by adding missing bracket.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -82,10 +82,16 @@
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of HtmlEncoder specifying characters to be encoded.</summary>
+        <returns>A new instance of the <see cref="HtmlEncoder"/></returns>
```
Set --> The set
HtmlEncoder class.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -82,10 +82,16 @@
+        <remarks><format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ Some characters in <paramref name="allowedRanges"/> might still get encoded, i.e. this parameter is just telling the encoder what ranges it is allowed to not encode, not what characters it must not encode.  
```
get encoded --> be encoded
is just telling...-> indicates what ranges the encoder is allowed to not encode, not what characters it must not encode.

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="JavaScriptEncoder"/> encodes, primarily which characters to encode.</param>
+        <summary>Creates a new instance of JavaScriptEncoder with provided settings.</summary>
```
of JavaScriptEncoder --> of the JavaScriptEncoder class
provided settings --> the specified settings

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="JavaScriptEncoder"/> encodes, primarily which characters to encode.</param>
+        <summary>Creates a new instance of JavaScriptEncoder with provided settings.</summary>
+        <returns>A new instance of the <see cref="JavaScriptEncoder"/>.</returns>
```
JavaScriptEncoder --> the JavaScriptEncoder class

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
```
Set --> The set

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of JavaScriptEncoder specifying characters to be encoded.</summary>
+        <returns>A new instance of the <see cref="JavaScriptEncoder"/>.</returns>
```
JavaScriptEncoder. --> JavaScriptEncoder class.

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -82,10 +82,15 @@
+        <remarks><format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ Some characters in <paramref name="allowedRanges"/> might still get encoded, i.e. this parameter is just telling the encoder what ranges it is allowed to not encode, not what characters it must not encode.  
```
get encoded --> be encoded
is just telling...-> indicates what ranges the encoder is allowed to not encode, not what characters it must not encode.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="HtmlEncoder"/> encodes, primarily which characters to encode.</param>
```
nit: used to control --> that control
nit: how the created HTMLEncoder encodes --> how the HtmlEncoder instance encodes
with provided -> with the provided

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="HtmlEncoder"/> encodes, primarily which characters to encode.</param>
+        <summary>Creates a new instance of HtmlEncoder with provided settings.</summary>
```
of HtmlEncoder, etc. --> of the HtmlEncoder class with the specified settings.

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="UrlEncoder"/> encodes, primarily which characters to encode.</param>
+        <summary>Creates a new instance of UrlEncoder with provided settings.</summary>
```
of UrlEncoder --> of the UrlEncoder class
with provided settings --> with the specified settings

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -103,9 +109,9 @@
-        <summary>To be added.</summary>
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets a default built-in instance of <see cref="HtmlEncoder"/>.</summary> 
```
HtmlEncoder --> the HtmlEncoder class

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -103,9 +108,9 @@
-        <summary>To be added.</summary>
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets a default built-in instance of <see cref="JavaScriptEncoder"/>.</summary>
```
JavaScriptEncoder --> the JavaScriptEncoder class

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -103,9 +108,9 @@
-        <summary>To be added.</summary>
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets a default built-in instance of <see cref="UrlEncoder"/>.</summary>
```
UrlEncoder --> the UrlEncoder class.

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -52,10 +52,10 @@
-        <remarks>To be added.</remarks>
+        <param name="settings">Settings used to control how the created <see cref="UrlEncoder"/> encodes, primarily which characters to encode.</param>
+        <summary>Creates a new instance of UrlEncoder with provided settings.</summary>
+        <returns>A new instance of the <see cref="UrlEncoder"/>.</returns>
```
UrlEncoder --> the UrlEncoder class

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of UrlEncoder specifying character to be encoded.</summary>
+        <returns>A new instance of the <see cref="UrlEncoder"/>.</returns>
```
UrlEncoder --> the UrlEncoder class

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -103,9 +109,9 @@
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets a default built-in instance of <see cref="HtmlEncoder"/>.</summary> 
+        <value>A <see cref="HtmlEncoder"/> object.</value>
```
A --> An

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -13,8 +13,9 @@
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary> An abstraction representing various text encoders.</summary>
```
An abstraction representing various text encoders --> The base class of web encoders.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -52,10 +53,10 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="value">String to encode.</param>
```
String --> The string

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -52,10 +53,10 @@
-        <remarks>To be added.</remarks>
+        <param name="value">String to encode.</param>
+        <summary>Encodes the supplied string and returns the encoded text as a new string.</summary>
+        <returns>>Encoded string.</returns>
```
Encoded --> The encoded

mairaw replied:
There's an extra `>` before the sentence that is making this sentence render as a note.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -77,10 +78,10 @@
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to this output.</param>
```
Better: The stream to which to write the encoded text.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -77,10 +78,10 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to this output.</param>
+        <param name="value">String to be encoded.</param>
```
String to be encoded. --> The string to encode.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -82,10 +82,16 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of HtmlEncoder specifying characters to be encoded.</summary>
```
to be encoded --> to encode

rpetrusha commented on xml/System.Text.Encodings.Web/JavaScriptEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of JavaScriptEncoder specifying characters to be encoded.</summary>
```
to be encoded --> to encode

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +105,12 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to the output.</param>
+        <param name="value">Array of characters to be encoded.</param>
```
Array --> The array
to be encoded --> to encode

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -133,12 +134,12 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to this output.</param>
+        <param name="value">String whose substring is to be encoded.</param>
```
String --> The string

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -168,11 +169,11 @@
-        <remarks>To be added.</remarks>
+        <param name="text">The text buffer to search.</param>
+        <param name="textLength">The number of characters in the <paramref name="text"/>.</param>
+        <summary>Finds index of the first character that needs to be encoded.</summary>
```
index --> the index
that needs to be encoded --> to encode

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -168,11 +169,11 @@
+        <param name="text">The text buffer to search.</param>
+        <param name="textLength">The number of characters in the <paramref name="text"/>.</param>
+        <summary>Finds index of the first character that needs to be encoded.</summary>
+        <returns>Index of the first character that needs to be encoded.</returns>
```
Index --> The index
that needs to be encoded --> to encode

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of UrlEncoder specifying character to be encoded.</summary>
```
UrlEncoder --> the UrlEncode class
character --> characters

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -52,10 +53,10 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="value">String to encode.</param>
+        <summary>Encodes the supplied string and returns the encoded text as a new string.</summary>
```
supplied --> specified

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -77,10 +78,10 @@
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to this output.</param>
+        <param name="value">String to be encoded.</param>
+        <summary>Encodes the supplied string into a <see cref="TextWriter"/>.</summary>
```
supplied --> specified
into a TextWriter --> to a TextWriter object

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +105,12 @@
-        <param name="characterCount">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to the output.</param>
```
Better: The stream to which to write the encoded text.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +105,12 @@
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to the output.</param>
+        <param name="value">Array of characters to be encoded.</param>
+        <param name="startIndex">The index where the substring starts.</param>
```
The starting index of the substring

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +105,12 @@
+        <param name="output">Encoded text is written to the output.</param>
+        <param name="value">Array of characters to be encoded.</param>
+        <param name="startIndex">The index where the substring starts.</param>
+        <param name="characterCount">Number of characters in the substring.</param>
```
Number --> The number

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +105,12 @@
+        <param name="value">Array of characters to be encoded.</param>
+        <param name="startIndex">The index where the substring starts.</param>
+        <param name="characterCount">Number of characters in the substring.</param>
+        <summary>Encodes characters from an array into a <see cref="TextWriter"/>.</summary>
```
into a TextWriter --> and writes them to a TextWriter object

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -133,12 +134,12 @@
-        <param name="characterCount">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="output">Encoded text is written to this output.</param>
```
Better: The stream to which to write the encoded text.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
+        <param name="buffer">The destination of the encoded text.</param>
+        <param name="bufferLength">Length of the destination <paramref name="buffer"/> in chars.</param>
```
Length --> The length
chars --> characters

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
```
Unicode scalar --> A Unicode scalar value.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
+        <param name="buffer">The destination of the encoded text.</param>
```
The destination of the encoded text --> A pointer to the buffer to which to write the encoded text

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
+        <param name="unicodeScalar">Unicode scalar.</param>
+        <param name="buffer">The destination of the encoded text.</param>
+        <param name="bufferLength">Length of the destination <paramref name="buffer"/> in chars.</param>
+        <param name="numberOfCharactersWritten">Number of characters written to the <paramref name="buffer"/>.</param>
```
Number --> The number

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
+        <param name="buffer">The destination of the encoded text.</param>
+        <param name="bufferLength">Length of the destination <paramref name="buffer"/> in chars.</param>
+        <param name="numberOfCharactersWritten">Number of characters written to the <paramref name="buffer"/>.</param>
+        <summary>Encodes a Unicode scalar into a buffer.</summary>
```
Encodes a Unicode scalar value and writes it to a buffer.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
+        <param name="bufferLength">Length of the destination <paramref name="buffer"/> in chars.</param>
+        <param name="numberOfCharactersWritten">Number of characters written to the <paramref name="buffer"/>.</param>
+        <summary>Encodes a Unicode scalar into a buffer.</summary>
+        <returns>Returns false if <paramref name="bufferLength"/> is too small to fit the encoded text, otherwise returns true.</returns>
```
text, otherwise returns
text; otherwise, returns

mairaw replied:
false -> `<see langword="false" />`
true -> `<see langword="true" />`

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +230,19 @@
+        <format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ This method is seldom called directly. One of the TextEncoder.Encode overloads should be used instead. Implementations of <see cref="TextEncoder"/> need to be thread safe and stateless.  
```
thread safe --> thread-safe

tompratt-AQ replied:
@rpetrusha I also changed:
 need to be --> must be
Is that correct?

rpetrusha replied:
@tompratt-AQ, yes, it's correct.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -261,10 +268,10 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
```
Unicode scalar --> A Unicode scalar value

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -261,10 +268,10 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
+        <summary>Determines if a given Unicode scalar will be encoded.</summary>
```
Unicode scalar --> Unicode scalar value

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -261,10 +268,10 @@
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">Unicode scalar.</param>
+        <summary>Determines if a given Unicode scalar will be encoded.</summary>
+        <returns>Returns true if the <paramref name="unicodeScalar"/> will be encoded by this encoder, otherwise returns false.</returns>
```
unicodeScalar --> unicodeScalar value
encoder, otherwise --> encoder; otherwise,

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -13,8 +13,8 @@
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Represents a filter which allows only certain Unicode code points through.</summary>
```
which --> that
allows only certain Unicode code points through --> allows only certain Unicode code points

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -49,9 +49,9 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="other">The other <see cref="TextEncoderSettings"/> to be cloned.</param>
+        <summary>Instantiates the filter by cloning the allow list of another <see cref="TextEncoderSettings"/>.</summary>
```
allow --> allowed list
TextEncoderSettings --> TextEncoderSettings object

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -223,8 +223,8 @@
       <Docs>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <summary>Resets this settings object by disallowing all characters.</summary>
```
this settings object --> this object

rpetrusha commented on xml/System.Text.Encodings.Web/UrlEncoder.xml:
```diff
@@ -82,10 +82,15 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">Set of characters that the encoder is allowed to not encode.</param>
```
Set --> The set

tompratt-AQ commented:
Terrific feedback Ron. Thanks for reviewing so closely. This was my first one of these, so I wasn't entirely sure how far I could go with changing the wording from the code comments. This will help me going forward.

mairaw commented:
Once you've addressed @rpetrusha's comments, please let me know. I want to find who owns this on the PU side for some technical review.

mairaw commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -30,8 +31,8 @@
       <Docs>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <summary>Initializes a new instance of the <see cref="TextEncoder"/> class.</summary>
```
these cref links in the summary need to have the full asset it, such as, T:System.Text.Encodings.Web.TextEncoder or it doesn't render as a link

mairaw commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -195,9 +196,9 @@
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets the maximum number of characters that this encoder can generate for each input character.</summary>
+        <value>Number of characters.</value>
```
The maximum number of characters.

mairaw commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -168,11 +169,11 @@
+        <param name="textLength">The number of characters in the <paramref name="text"/>.</param>
+        <summary>Finds index of the first character that needs to be encoded.</summary>
+        <returns>Index of the first character that needs to be encoded.</returns>
+        <remarks>This method is seldom called directly. It's used by higher level helper APIs.</remarks>
```
I'd probably change this to be more actionable to the customer. Something like You don't usually call this method directly.

mairaw commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -352,9 +352,9 @@
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Gets an enumeration of all allowed code points.</summary>
```
an enumeration -> an enumerator

mairaw reviewed (changes requested):
Left some comments in addition to the ones @rpetrusha already gave you. Some apply to more than one place but I only said once.

tompratt-AQ committed:
Made corrections per review comments.

tompratt-AQ commented:
@mairaw This seems to have failed the check with no coding error, just: "Run script failed with no detailed information."

mairaw commented:
Let's see if it happens again.

tompratt-AQ commented:
Happened again with the same error.

mairaw commented:
Trying one more time, I'll open a LSI if it doesn't work. There are other builds failing.

tompratt-AQ commented:
Thanks. Yes, it looks like PR 2226 failed with the same error.

tompratt-AQ commented:
@mairaw Thanks for starting this again last night!

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -13,8 +13,8 @@
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Represents a type used to do HTML encoding.</summary>
```
It can't represent a type, since it is a type. So better, "Represents an HTML character encoding." This comment applies to the other types as well.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -82,10 +82,16 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">The set of characters that the encoder is allowed to not encode.</param>
+        <summary>Creates a new instance of the HtmlEncoder class specifying characters to encode.</summary>
```
specifying --> that specifies

It appears, though, that the parameter is not characters to encode, but ranges the encoder is allowed to not encode. So perhaps it would be better to write "characters the encoder is allowed to not encode."

This comment also applies to the other encoding Default properties.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -82,10 +82,16 @@
+        <remarks><format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ Some characters in `allowedRanges` might still be encoded, that is, this parameter indicates what ranges the encoder is allowed to not encode, not what characters it must not encode.  
```
encoded; that is, 
Again, this comment applies to all the encoder Create(UnicodeRange[]) methods.

rpetrusha commented on xml/System.Text.Encodings.Web/HtmlEncoder.xml:
```diff
@@ -103,9 +109,9 @@
-        <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <summary>Gets a default built-in instance of the <see cref="T:System.Text.Encodings.Web.HtmlEncoder"/> class.</summary> 
+        <value>A default built-in instance of the <see cref="T:System.Text.Encodings.Web.HtmlEncoder"/> class.</value>
```
It seems redundant to say it returns the default built-in instance. It's OK to say it returns the default instance, but in that case we should specify in remarks what the defaults are (in the source, it's indicated by the internal DefaultHtmlEncoder class); otherwise, built-in instance is best.

This also is true of the other encoder Default properties.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -13,8 +13,14 @@
+    <format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ TextEncoder subclasses can be used to do HTML encoding, URI encoding, and JavaScript encoding. Instances of such subclasses can be accessed using the <xref:System.Text.Encodings.Web.HtmlEncoder.Default>, <xref:System.Text.Encodings.Web.UrlEncoder.Default>, and <xref:System.Text.Encodings.Web.JavaScriptEncoder.Default> methods.  
```
Each of these xrefs needs to include `?displayProperty=fullName`
Default methods --> Default properties

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +110,12 @@
-        <remarks>To be added.</remarks>
+        <param name="output">The stream to which to write the encoded text.</param>
+        <param name="value">The array of characters to encode.</param>
+        <param name="startIndex">The starting index of the substring.</param>
```
There is no substring. So: The array index of the first character to encode.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -104,12 +110,12 @@
+        <param name="output">The stream to which to write the encoded text.</param>
+        <param name="value">The array of characters to encode.</param>
+        <param name="startIndex">The starting index of the substring.</param>
+        <param name="characterCount">The number of characters in the substring.</param>
```
The number of characters in the array to encode.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -168,11 +174,17 @@
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="text">The text buffer to search.</param>
+        <param name="textLength">The number of characters in the <paramref name="text"/>.</param>
```
delete "the" before the paramref

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +241,19 @@
+        <param name="unicodeScalar">A Unicode scalar value.</param>
+        <param name="buffer">A pointer to the buffer to which to write the encoded text.</param>
+        <param name="bufferLength">The length of the destination <paramref name="buffer"/> in characters.</param>
+        <param name="numberOfCharactersWritten">The number of characters written to the <paramref name="buffer"/>.</param>
```
Since this is an out param, change this to, "When the method returns, indicates the number of characters..."

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +241,19 @@
+        <format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ This method is seldom called directly. One of the TextEncoder.Encode overloads should be used instead. Implementations of the <xref:System.Text.Encodings.Web.TextEncoder> class must be thread-safe and stateless.  
```
make TextEncoder.Encode an xref.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -261,10 +279,10 @@
-        <remarks>To be added.</remarks>
+        <param name="unicodeScalar">A Unicode scalar value.</param>
+        <summary>Determines if a given Unicode scalar value will be encoded.</summary>
+        <returns>Returns <see langword="true" /> if the <paramref name="unicodeScalar"/> value will be encoded by this encoder; otherwise, returns <see langword="false" />.</returns>
```
We don't need to include "returns" in the return value description. So, "true if the unicodeScalar value...; otherwise, false.
This applies to all methods with a Boolean return value.

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -49,9 +49,9 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="other">The other <see cref="T:System.Text.Encodings.Web.TextEncoderSettings"/> object to be cloned.</param>
+        <summary>Instantiates the filter by cloning the allowed list of another <see cref="T:System.Text.Encodings.Web.TextEncoderSettings"/> object.</summary>
```
nit: the filter --> a filter

rpetrusha commented on xml/System.Text.Encodings.Web/TextEncoderSettings.xml:
```diff
@@ -75,9 +75,9 @@
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="allowedRanges">The allowed character ranges.</param>
+        <summary>Instantiates the filter where only the character ranges specified by <paramref name="allowedRanges"/> are allowed by the filter.</summary>
```
nit: the filter --> a filter

tompratt-AQ committed:
Made changes to address review comments.

rpetrusha commented:
@tompratt-AQ, I think that this looks good. @mairaw, do you want to review it? I think that it's ready for PU review, which we should do before we merge.

rpetrusha commented:
@KrzysztofCwalina and @tarekgh, we'd appreciate it if you could review this.

tarekgh commented:
LGTM. 

@KrzysztofCwalina please have a look too. 

CC @krwq

mairaw commented:
For easier review, you can the content on staging too: https://review.docs.microsoft.com/en-us/dotnet-new/api/System.Text.Encodings.Web?branch=pr-en-us-2221&view=netcore-2.0

@tompratt-AQ we should also add the namespace description at https://github.com/dotnet/docs/blob/master/xml/ns-System.Text.Encodings.Web.xml

mairaw commented on xml/System.Text.Encodings.Web/TextEncoder.xml:
```diff
@@ -229,13 +241,19 @@
+        <format type="text/markdown"><![CDATA[ 
+
+## Remarks  
+ This method is seldom called directly. One of the <xref:System.Text.Encodings.Web.TextEncoder.Encode> overloads should be used instead. Implementations of the <xref:System.Text.Encodings.Web.TextEncoder> class must be thread-safe and stateless.  
```
this xref link is broken

mairaw reviewed (approved):
LGTM. Only one broken link and the namespace description to be added.

tarekgh commented:
@mairaw thanks for the link it is really much easier to look at there. 

In Text​Encoder.​Max​Output​Characters​Per​Input​Character, it mentioned "Gets the maximum number of characters that this encoder can generate for each input character."
I would change the last part "input character" to "input code point". 

I am looking at the rest now.

tarekgh commented:
side question, why the docs not mentioning the possible exception get thrown for the APIs?

mairaw commented:
We should add those too @tarekgh. We'll work with @tompratt-AQ to add those. Thanks!

tarekgh commented:
I looked at the rest and LGTM. 

one last question, is it possible to point to the System.Text.Encodings.Web project in github for looking at the source also use the tests there as a code samples? 

Thanks all for getting this done!

rpetrusha commented:
@tarekgh, yes, it is possible. I'll ask @tompratt-AQ to do it next week.

tompratt-AQ commented:
Thanks all for the great review comments. I'll address these issues next week.

tompratt-AQ committed:
Added Namespace description, exceptions, and adressed review comments.

tompratt-AQ commented:
@rpetrusha In the most recent commit, I addressed the following:

1. Broken link that @mairaw mentioned.
2. Change to Text​Encoder.​Max​Output​Characters​Per​Input​Character per takegh.
3. Added a Namespace description
4. Added exceptions for each class based on the code files.

Please especially review my namespace description and my exception information as this is the first time I've added either. 
Especially, look at the exceptions in the System.Text.Encodings.Web.TextEncoder base class. ([code file ](https://github.com/dotnet/corefx/blob/master/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs))
 I wasn't entirely sure of the best way to note the exceptions thrown by private classes which are called by the public classes. Thanks!

tompratt-AQ committed:
Fixed errors with cref closing tags.

rpetrusha commented:
@tompratt-AQ, the final item that @tarekgh had wanted was a link to source and to examples (as found in unit tests). In the type topic for each encoder type and for TextEncoderSettings, add a paragraph at the  end of the Remarks section that reads:

The source code for this type is available in the [System.Text.Encodings.Web project](https://github.com/dotnet/corefx/srx/System.Text.Encodings.Web/src/) on GitHub. Unit tests that can also serve as code examples are found in the [System.Text.Encodings.Web/tests](https://github.com/dotnet/corefx/srx/System.Text.Encodings.Web/src/) folder on GitHub.

tarekgh commented:
@rpetrusha the links you mentioned in your last comment are invalid. the github link for this library project is https://github.com/dotnet/corefx/tree/master/src/System.Text.Encodings.Web

tompratt-AQ committed:
Added paragraph with links to code and tests in Remarks for each type.

tompratt-AQ commented:
@rpetrusha I added your suggested paragraph to the remarks section for each type, and confirmed that the links were correct.

rpetrusha commented:
Ah, of course, @tarekgh. I always seem to create invalid GitHub links. Fortunately, you noticed, and @tompratt-AQ seems to have added the correct links to the documentation. If this builds successfully and the links are OK, we'll merge.
